pkg/operator/controller/dns_status_test.go

```diff
@@ -547,39 +670,97 @@ func TestSkippingStatusUpdates(t *testing.T) {
+				Requeue:      true,
+				RequeueAfter: 5 * time.Second,
+			},
 		},
 		{
-			name:      "no clusterIP, would return Progressing=ConditionTrue, but Degraded was set to false within tolerated duration, so returns Progressing=ConditionFalse",
-			clusterIP: "",
+			name:         "should return Degraded=ConditionFalse because not enough time has elapsed for an update",
+			clusterIP:    "1.2.3.4",
+			dnsDaemonset: makeDaemonSet(6, 0, 6),
+			nrDaemonset:  makeDaemonSet(6, 6, 6),
 			oldCondition: operatorv1.OperatorCondition{
 				Type:               operatorv1.OperatorStatusTypeDegraded,
 				Status:             operatorv1.ConditionFalse,
-				LastTransitionTime: metav1.NewTime(time.Date(2022, time.Month(5), 19, 1, 9, 50, 0, time.UTC)),
+				LastTransitionTime: metav1.NewTime(time.Date(2022, time.Month(5), 12, 1, 10, 50, 0, time.UTC)),
+			},
+			currentTime: time.Date(2022, time.Month(5), 12, 1, 10, 52, 0, time.UTC),
+			// last - current = 2s, and we tolerate 5s, so this should not change from False.
+			toleration: 5 * time.Second,
+			expected:   operatorv1.ConditionFalse,
+		},
+		{
+			name:         "should return Degraded=ConditionFalse because it is Progressing",
+			clusterIP:    "1.2.3.4",
+			dnsDaemonset: makeDaemonSet(6, 0, 6),
+			nrDaemonset:  makeDaemonSet(6, 6, 6),
+			oldCondition: operatorv1.OperatorCondition{
+				Type:               operatorv1.OperatorStatusTypeDegraded,
+				Status:             operatorv1.ConditionFalse,
+				LastTransitionTime: metav1.NewTime(time.Date(2022, time.Month(5), 12, 1, 10, 50, 0, time.UTC)),
+			},
+			progressingCondition: operatorv1.OperatorCondition{
+				Type:   operatorv1.OperatorStatusTypeProgressing,
+				Status: operatorv1.ConditionTrue,
 			},
 			currentTime: time.Date(2022, time.Month(5), 19, 1, 10, 50, 0, time.UTC),
-			// last-curr = 1m, tolerate 1m, so should prevent the flap.
-			toleration: 1 * time.Minute,
+			toleration:  5 * time.Second,
+			expected:    operatorv1.ConditionFalse,
+		},
+		{
+			name:         "would return Degraded=ConditionTrue, but Degraded was set to false within tolerated duration, so returns Degraded=ConditionFalse",
+			clusterIP:    "1.2.3.4",
+			dnsDaemonset: makeDaemonSet(6, 1, 6),
+			nrDaemonset:  makeDaemonSet(6, 6, 6),
+			oldCondition: operatorv1.OperatorCondition{
+				Type:   DNSMaxUnavailableDNSPodsExceeded,
+				Status: operatorv1.ConditionFalse,
+				//LastTransitionTime: metav1.NewTime(time.Date(2022, time.Month(5), 19, 1, 9, 50, 0, time.UTC)),
+				// 1 minute ago
+				LastTransitionTime: metav1.NewTime(clock.Now().Add(time.Minute * -1)),
+			},
+			//currentTime: time.Date(2022, time.Month(5), 19, 1, 10, 50, 0, time.UTC),
+			currentTime: metav1.NewTime(clock.Now()).Time,
+			// last-curr = 1m, tolerate 2m, so should prevent the flap.
+			toleration: 2 * time.Minute,
 			expected:   operatorv1.ConditionFalse,
+			reconcileResult: reconcile.Result{
+				Requeue:      true,
+				RequeueAfter: 2 * time.Minute,
+			},
```

Comment:
Regarding my comment about `GracePeriod` not working as expected, we should probably have a unit test for after the `GracePeriod` expires for the `DNSMaxUnavailableDNSPodsExceeded` condition, it should go degraded.

It may be possible to write a successful unit test for this by manipulating the clock, but my concern is that wouldn't be possible in real life.

Reply:
No more GracePeriod.  Stay tuned for updates.